### PR TITLE
🐛 fix: make build creating no categories page

### PR DESCRIPTION
**Pull request by gwainor**
image

### CODE REVIEW

File: src/app/categories/[slug]/page.tsx [hunk 1]
```diff
@@ -25,7 +24,7 @@ export default function CategoryPage({ params: { slug } }: Props) {
     );
 }
 
-export async function generateStaticParams() {
+export function generateStaticParams() {
```
Comment (gwainor) [inline src/app/categories/[slug]/page.tsx:27]: # Reviewer Note
It does not has to be `async` if you are not doing any API call

File: src/app/categories/[slug]/page.tsx [hunk 2]
```diff
@@ -1,4 +1,3 @@
-'use client';
```
Comment (gwainor) [inline src/app/categories/[slug]/page.tsx:1]: # Reviewer Note
This was the problematic statement causing dynamic page generation.

File: src/constants/content.ts [hunk 3]
```diff
@@ -1,20 +1,29 @@
...
-    CATEGORIES_BY_SLUG[category.slug] = category;
-});
+export const CATEGORIES_BY_SLUG: { [key: string]: Category } =
+    CATEGORIES.reduce(
+        (collection, category) => ({
+            ...collection,
+            [category.slug]: category,
+        }),
+        {}
+    );
```
Comment (gwainor) [inline src/constants/content.ts:16]: # Reviewer Note
Idiomatic way of JavaScript.

File: src/components/CategoryBooks/index.tsx [hunk 4]
```diff
@@ -13,9 +14,6 @@ type Props = {
 
 export function CategoryBooks({ category }: Props) {
     const bookIds: string[] = lookUpBookIdsByCategory(category);
-    if (typeof bookIds === 'undefined') {
-        notFound();
-    }
```
Comment (gwainor) [inline src/components/CategoryBooks/index.tsx:18]: # Reviewer Note
This moved to the page component.

File: src/components/CategoryBooks/index.tsx [hunk 5]
```diff
@@ -1,3 +1,4 @@
+'use client';
```
Comment (gwainor) [inline src/components/CategoryBooks/index.tsx:1]: # Reviewer Note
React hooks does not work without this.

File: src/app/categories/[slug]/page.tsx (hunk 2 above)
Comment (jontsai) [reply src/app/categories/[slug]/page.tsx:1]: That's what I also thought at first.

**Review by jontsai — APPROVED** [review]
LGTM

File: src/constants/content.ts (hunk 3 above)
Comment (jontsai) [reply src/constants/content.ts:16]: I'm not sure, I feel like `...collection` will actually make this an `O(n^2)` operation, instead of `O(n)`

**Pull request merged**